### ISSUE TITLE
No snake

### ISSUE THREAD

**Issue by SlimeDog** [MysticCity/MinigamesAPI#31]
Spigot 1.10 latest (yesterday) with a standard complement of plugins: WE/WG, MV, PEX, etc.
MinigamesLib and MGSnake dev build 20.

For testing, I configured for 1-4 players. I started the game. There were slimes and footprints, but no snake following behind me. I assume that there should be.

**Comment by mepeisen** [comment]
Can you check your plugins?
I used the minigames-all build and only plugin I used was essentialsX. There were sheeps and a snake. Anyhow the sheeps still are walking not very fine, are looking in wrong direction etc. This is already covered with Issue #4. Seems to be some timing problems since 1.9.

On our life server (1.9.4) we got a similar problem. WorldGuard seems to interrupt the sheep creation and there were no sheeps. Can you review your WorldGuard flags? As soon as we allowed spawning sheeps snake worked again.

However I think this can be classified as a bug.
Similar to the soft requirement of essentials to let snake work at all we should investigate how to work around world guards flags. For example: Maybe we can detect if world guard is running and warn the admin that sheeps are disabled if the flags are wrong. As world guard is a popular plugin this would be fair.

**Comment by SlimeDog** [comment]
Good catch. I was blocking sheep spawn with WG. Fixed, reloaded. Still no sheep. :(

Agreed that you should work around the spawn issue, in any event. Just in the arena and just during games.

**Comment by mepeisen** [comment]
for us it was the following flags:
region flag -w world __global__ damage-animals allow
region flag -w world __global__ mob-spawning allow

**Comment by SlimeDog** [comment]
I must have a more restrictive regions.yml. When I removed all of it, I see sheep. I also see slimeballs, which I didn't before, and my boots dropped at some point.

So, the sheep are the snake? Can we have wool back as an option?

**Comment by mepeisen** [comment]
fixed sheep yaws and did some refactoring. Wool is back again, you can use them with per arena config option "useSheeps: false".
However the wool does not yet work. Please watch this bug for changes.

**Comment by mepeisen** [comment]
Finally I did some cleanup and bugfixing.
For direct downloading: http://nexus.xworlds.eu/nexus/content/repositories/mce-snapshots/com/github/MCE-Plugins/MGSnake-Plugin/1.14.9-SNAPSHOT/MGSnake-Plugin-1.14.9-20160627.105509-5.jar
Seems that this is more stable now.

And we have two new config options (per arena):
- useSheeps: true|false
- tickCount: <integer>

The use sheeps switches between sheeps and classic wool.

The tick count is a tick for remembering snake positions (defaults to 2). The larger the value the longer a snake will be. Setting to 1 will make the snake more "solid" but letting the sheeps and the wool be more like a lump. I did not test it with values larger than 4.

The block/sheep yaw still seems to be corrupt.
